_posts/2020-06-21-one-stop-guide-to-lombok.md

```diff
@@ -0,0 +1,337 @@
+
+```java
+<dependencies>
+  <dependency>
+    <groupId>org.projectlombok</groupId>
+    <artifactId>lombok</artifactId>
+    <version>1.18.12</version>
+    <scope>provided</scope>
+  </dependency>
+</dependencies>
+```
+
+*N.B.* ~when adding lombok annotations make sure that you have imported them from correct package. This is the correct import for annotation `@Getter`  
+`import lombok.Getter`~
+
+If you use Jdk9+ you should add the following to the maven compiler plugin configuration section:  
+```java
+<annotationProcessorPaths>
+  <path>
+    <groupId>org.projectlombok</groupId>
+    <artifactId>lombok</artifactId>
+    <version>1.18.12</version>
+  </path>
+</annotationProcessorPaths>
+```
+
+## Dependencies to add for Gradle
+
+As with setting up lombok for Maven you have to specify lombok dependency in your `buiild.gradle`
+
+In case you do not want to use Lombok Gradle plugin gradle has the built-in `compileOnly` scope, which can be used to tell Gradle to add lombok only during compilation.
+
+```java
+dependencies {
+  compileOnly 'org.projectlombok:lombok:1.18.12'
+  annotationProcessor 'org.projectlombok:lombok:1.18.12'
+}
+```
+
+If you want to use lombok gradle plugin you can read more about it [https://plugins.gradle.org/plugin/io.freefair.lombok](here)
+
+## Settings for IntelliJ Idea to activate Lombok preprocessing
+
+To use Lobmok features you have to activate preprocessing of Lombok annotations. For Intellij you have to go to  
+~`Settings > Build, Execution, Deployment > Compiler > Annotation Processors`~ and activate `Enable annotation processing`
+
+The next popular features is `@EqualsAndHashCode`. With this feature usually `@EqualsAndHashCode.Exclude` used to exclude some fields from using in `equals()` and `hashCode()` methods.
+
+# Caveats when using some features of Lombok
+
+Using Lombok annotations is easy. But sometimes there are some caveats. They often can be met when you have extended from some superclass.
+In those cases you can use some additional parameter like `callSuper=true`. This parameter can be used in such features like `@ToString, @EqualsAndHashCode etc.`. Using or not the parameter `callSuper` can be not such harmful in `@ToString` as it can be in `@EqualsAndHashCode`.
+Also it's a best practice to manually exclude all the fields you do not want to be used by Lombok when generating the code. This can be done by `@ToString.Exclude` ot `@EqualsAndHashCode.Exclude` annotations.
+
+
+# DeLombok
+
+Lombok helps to focus you on the code you write and not to focus on creating typical constructors, setters, getters etc. It removes *a lot* of boilerplate code.
+But it doesn't cover cases. You cannot plug lombok into javadoc or other tools which use java sources. Here you can use delombok. Delombok copies an entire directory into another directory, recursively, skiping class files and applying lombok transformations to any java source files it encounters.
```

Comment:
I don't understand the cases it doesn't cover. Please explain "plugging Lombok into Javadoc" and add some example for "tools which use java sources".